missilewars-plugin/src/main/java/de/butzlabben/missilewars/listener/PlayerListener.java

```diff
@@ -92,117 +84,125 @@ public void onPickup(PlayerPickupItemEvent event) {
-            event.setCancelled(true);
-    }
+        if (!onJoinGame.isCancelled()) {
+            game.updateGameInfo();
+            sendEventDebugMessage(player, game);
+            Logger.NORMAL.log(player.getName() + " joint the MW game " + game.getLobby().getName());
+        } else {
+            Logger.DEBUG.log("Canceling game join for " + player.getName());
+        }
 
-    @EventHandler(priority = EventPriority.MONITOR, ignoreCancelled = true)
-    public void onPreJoin(PrePlayerArenaJoinEvent event) {
-        Logger.DEBUG.log("PrePlayerArenaJoinEvent: " + event.getPlayer().getName());
-        Bukkit.getScheduler().runTask(MissileWars.getInstance(), () -> Bukkit.getPluginManager()
-                .callEvent(new PlayerArenaJoinEvent(event.getPlayer(), event.getGame())));
+        return onJoinGame;
     }
 
-    // Debugging staff
+    private PlayerArenaLeaveEvent registerPlayerArenaLeaveEvent(Player player, Game game) {
+        PlayerArenaLeaveEvent onLeaveGame = new PlayerArenaLeaveEvent(player, game);
+        Bukkit.getPluginManager().callEvent(onLeaveGame);
 
-    @EventHandler
-    public void onPlayerArenaJoin(PlayerArenaJoinEvent event) {
-        Logger.DEBUG.log("PlayerArenaJoinEvent: " + event.getPlayer().getName());
-    }
+        game.updateGameInfo();
+        sendEventDebugMessage(player, game);
+        Logger.NORMAL.log(player.getName() + " left the MW game " + game.getLobby().getName());
 
-    @EventHandler
-    public void onPlayerArenaLeave(PlayerArenaLeaveEvent event) {
-        Logger.DEBUG.log("PlayerArenaLeaveEvent: " + event.getPlayer().getName());
+        return onLeaveGame;
     }
 
+    /**
+     * This method gets the game based of the location. It's either inside
+     * the game lobby (representing as an area) or inside the game arena
+     * (representing as a world).
+     *
+     * @param location (Location) of the player
+     * @return the Game Object if existing for the location
+     */
     private Game getGame(Location location) {
         if (GameManager.getInstance() == null) return null;
 
         return GameManager.getInstance().getGame(location);
     }
 
-    /**
-     * Checks if cancelled and spits out events
-     */
-    private boolean checkJoinOrLeave(Player player, Game from, Game to) {
-        if (to != null && to != from) {
-            PrePlayerArenaJoinEvent event = new PrePlayerArenaJoinEvent(player, to);
-            Bukkit.getPluginManager().callEvent(event);
-            if (event.isCancelled()) player.sendMessage(MessageConfig.getMessage("not_enter_arena"));
-            return event.isCancelled();
-        }
-        if (from != null && to != from) {
-            PlayerArenaLeaveEvent event = new PlayerArenaLeaveEvent(player, from);
-            Bukkit.getPluginManager().callEvent(event);
-        }
-        return false;
-    }
+    private void sendEventDebugMessage(Player player, Game game) {
```

Comment:
Maybe rename to logEventDebugMessage, but good idea to have more possibilities for debugging :+1: 

```suggestion
    private void logEventDebugMessage(Player player, Game game) {
```

Reply:
I don't know. You don't "log" the debug message. You send it.

Reply:
Well we literally do

```java
        Logger.DEBUG.log("Location: " + player.getLocation());
        Logger.DEBUG.log("Current game amount: " + GameManager.getInstance().getGameAmount());
        Logger.DEBUG.log("Lobby: " + game.getLobby().getDisplayName());
        Logger.DEBUG.log("Arena: " + game.getArena().getDisplayName());
        Logger.DEBUG.log("Team 1: " + game.getTeam1());
        Logger.DEBUG.log("Team 2: " + game.getTeam2());
```